src/event-progress-bar.js

```diff
@@ -0,0 +1,63 @@
+// Update the count down every 1 second
+var x = setInterval( doProgress, 1000 );
+
+function doProgress() {
+
+	/**
+	 * Get all the active progress bars from the page.
+	 *
+	 * Child elements:
+	 * 0 - progress-bar-container__live
+	 * 1 - progress-bar-container__background
+	 * 2 - progress-bar-container__timeleft
+	 */
+	var events = document.getElementsByClassName( 'progress-bar-container__on' );
+
+	// Bail if there are no live events.
+	if ( events.length <= 0 ) {
+		//clearInterval( x );
+		return;
+	}
+
+	// Get today's date and time and the offset from UTC.
+	var d = new Date();
+	var now = d.getTime();
+	var tzOffset = d.getTimezoneOffset();
+	var offset = tzOffset * 60 * 1000;
```

Comment:
Not all time zones are hours. Some are 30min or 45min.
Why are we doing this at all? (e.g. `timeLeft = endDate - (now + offset);`)
Why not use [MomentJS](https://momentjs.com/), as provided by TEC?

Reply:
We are getting the events start and end times in UTC time.

`now` gives you the current time in your own timezone.
The `offset` will give you the offset from UTC in seconds.
Later down we use `(now + offset)` to get the current UTC time.

This way everything (event times and the current time) is in UTC, so the calculations will be right.

FYI: I did not change anything in the code.